### PR TITLE
[10.0][MIG] purchase_date_planned_manual

### PR DESCRIPTION
**Pull request by MiquelRForgeFlow**
Standard migration.

### CODE REVIEW

**Review by elicoidal — APPROVED** [review]

File: purchase_date_planned_manual/__openerp__.py
Comment (LoisRForgeFlow) [inline purchase_date_planned_manual/__openerp__.py]: rename to `__manifest__.py` please

**Comment by LoisRForgeFlow** [comment]
I just realized of one issue:
1. Create procurement for imac with scheduled date July 1st. PO assigned.
2. Create procurement for imac with scheduled date July 7th. Same PO is assigned but the line is merged with the previous one (so scheduled date in PO will be July 1st).

cc @jbeficent

**Comment by MiquelRForgeFlow** [comment]
@lreficent , to fix this issue you mentioned, we would need to add the full method `make_po` defined in `purchase` module with a minor change, but this method is too long.

May Odoo accepts to improve the method: https://github.com/odoo/odoo/pull/18467

File: purchase_date_planned_manual/models/purchase_order.py
Comment (congdpt) [inline purchase_date_planned_manual/models/purchase_order.py]: All python native imports first, then odoo
See OCA contributing guidelines

File: purchase_date_planned_manual/tests/test_date_planned_manual.py
Comment (congdpt) [inline purchase_date_planned_manual/tests/test_date_planned_manual.py]: All python native imports first, then odoo
See OCA contributing guidelines

**Review by congdpt — CHANGES_REQUESTED** [review]

**Comment by AaronHForgeFlow** [comment]
@mreficent Maybe making this module depending on https://github.com/OCA/purchase-workflow/tree/9.0/procurement_purchase_no_grouping (after its migration) would solve the issue commented by @lreficent

**Comment by lasley** [comment]
Any updates on this?

**Comment by LoisRForgeFlow** [comment]
@lasley We are still waiting some reaction [here](https://github.com/odoo/odoo/pull/18467), but maybe we have to start thinking about moving forward w/out it :sweat_smile:

**Comment by lasley** [comment]
@lreficent - Thanks for the update. Maybe we'll give that one a few more days now that we've pinged on it. Realistically though, it's an IMP, so I don't think the likelihood of merge on a production branch is high.

**Comment by LoisRForgeFlow** [comment]
@mreficent @lasley As they don't agree to add the hook... which do you think would be the better approach?

**Comment by lasley** [comment]
@lreficent What about https://github.com/OCA/purchase-workflow/pull/422#issuecomment-328048901?

**Comment by LoisRForgeFlow** [comment]
@lasley @aheficent Not sure, that could be a functional fix for sure, but I don't like the idea to add `procurement_purchase_no_grouping` as a dependency and use a script to add the 'No line grouping' option to all product groups. Also, I would be ok to merge lines with same planned date I think.

**Review by hparfr — APPROVED** [review]
Thanks

File: purchase_date_planned_manual/models/purchase_order.py
Comment (pedrobaeza) [inline purchase_date_planned_manual/models/purchase_order.py]: Use `fields.Datetime.from_string()` instead

**Review by pedrobaeza — CHANGES_REQUESTED** [review]

File: purchase_date_planned_manual/models/purchase_order.py
Comment (MiquelRForgeFlow) [reply purchase_date_planned_manual/models/purchase_order.py]: done

**Review by LoisRForgeFlow — APPROVED** [review]
:+1:

**Review by bealdav — APPROVED** [review]

**Review by congdpt — APPROVED** [review]

File: purchase_date_planned_manual/models/purchase_order.py
Comment (rousseldenis) [inline purchase_date_planned_manual/models/purchase_order.py]: @mreficent You can remove as https://github.com/odoo/odoo/commit/a572b8115cdaa3a28f5c977a6006ddfa95647987

File: purchase_date_planned_manual/models/purchase_order.py
Comment (rousseldenis) [inline purchase_date_planned_manual/models/purchase_order.py]: @mreficent Please put field definition on top and quote compute method

**Review by rousseldenis — CHANGES_REQUESTED** [review]
Code review

File: purchase_date_planned_manual/models/purchase_order.py
Comment (MiquelRForgeFlow) [reply purchase_date_planned_manual/models/purchase_order.py]: why? aesthetic change?

Comment (rousseldenis) [reply purchase_date_planned_manual/models/purchase_order.py]: No I'm not so mad :smile:  but because of https://github.com/OCA/maintainer-tools/blob/master/CONTRIBUTING.md#models

**Comment by rousseldenis** [comment]
By the way I will propose a PR on Odoo to recompute date_planned on lines if date_order is changed (onchange) on draft PO and if purchase lines have no procurement.

**Comment by rousseldenis** [comment]
https://github.com/odoo/odoo/pull/26697

**Comment by MiquelRForgeFlow** [comment]
@rousseldenis we should wait odoo feedback of your PR or we could merge this PR as it has already 5 approvals?

**Comment by rousseldenis** [comment]
Of course not. It was for information purpose. We can move on this

**Review by rousseldenis — APPROVED** [review]

**Comment by hparfr** [comment]
Is it ok for the merge @pedrobaeza  ? Thanks

**Review by pedrobaeza — APPROVED** [review]

**Pull request merged**
